edx_event_bus_kafka/internal/consumer.py

```diff
@@ -132,6 +133,24 @@ def _shut_down(self):
         """
         self._shut_down_loop = True
 
+    def _db_needs_reconnect(self):
+        """
+        Determine if we need to try to reconnect to the database
+
+        Returns:
+            True if we have a db connection but it is unusable
+            False otherwise (if we don't have a connection or our current connection is fine)
+        """
+        return connection.connection and not connection.is_usable()
```

Comment:
1. Checking `connection.connection` first might be totally necessary, but seems really odd. Should we link to docs? Does it raise an exception if you call `connection.is_usable()` without checking this? If it isn't needed, I think we should drop it.
2. [Nit] If it is needed, I wonder if the following would make it a little more readable:
```
has_connection = bool(connection.connection)
return has_connection and not connection.is_usable()
```

Reply:
`connection.is_usable()` does indeed raise if connection.connection is null. I'm pretty ambivalent as to the structure so will use your suggestion.

Reply:
[fixed]